WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadService.java

```diff
@@ -56,6 +56,15 @@
     @Inject PostStore mPostStore;
     @Inject SiteStore mSiteStore;
 
+    private class UploadingPost {
+        PostModel postModel;
+        List<MediaModel> pendingMedia = new ArrayList<>();
+        UploadingPost(PostModel postModel, List<MediaModel> pendingMedia) {
+            this.postModel = postModel;
+            this.pendingMedia = pendingMedia;
+        }
+    }
+
```

Comment:
Using a `<Integer, List<MediaModel>>` map is clear enough, the name of the var suggests it's a map of `Posts-to-something`, it occupies less memory and we're otherwise keeping a whole model in memory, when the id is all we need. If anyone wants to obtain something from the model tomorrow, they might think "ah I've got the model right here, great, I'll just use it!" only to get into some obscure bug because they might be reading an old copy.
My take is: do you happen to _need_ model? Get the freshest copy from the DB, right when you need it. We don't seem to be needing it in this PR, right?

Also if we are using another way to do this already (like the `sCompletedMediaByPost` I pointed out), it's clearer if we keep the same pattern to "keep something for a post" so a future developer looking at the code doesn't need to figure out the differences. Even if we decided to keep the `UploadingPost` class, it'd be better if we also have a similar class for the rest of cases where this kind of list/map is needed.
I still think the postID approach in this very case is enough, we don't need the rest.

For a pragmatic approach to move forward: we have 2 options: either we keep the id only, in this map, dropping the new `UploadingPost` class, or we also update the `sCompletedMediaByPost`  to have a class of its own that represents stuff in a similar way.

I still think using the id is enough, for the reasons outlined above, but I'm not stopping this from being merged for this anyway

What do you think?